src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../PeopleFilters';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { filteredPeople } from '../../utils/flteredPeopleList';
+
+export const PeoplePage = () => {
+  const [peopleList, setPeopleList] = useState<Person[] | null>(null);
+  const [isError, setIsError] = useState();
+  const [isLoad, setIsLoad] = useState(false);
+  const [searchParams] = useSearchParams();
+
+  const listOfPeople = () => (peopleList ? [...peopleList] : []);
+
+  const sortParam = searchParams.get('sort');
+  const orderParam = searchParams.get('order');
+
+  const getPeopleList = () => {
+    setIsLoad(true);
+
+    getPeople()
+      .then((data) => setPeopleList(data))
+      .catch((error) => setIsError(error.message))
+      .finally(() => setIsLoad(false));
+  };
+
+  const sortPerson = (a: Person, b: Person) => {
+    switch (sortParam) {
+      case 'name': return a.name.localeCompare(b.name);
+      case 'sex': return a.sex.localeCompare(b.sex);
+      case 'born': return a.born - b.born;
+      case 'died': return a.died - b.died;
+
+      default: return 0;
+    }
+  };
+
+  const getSortedPeopleList = (): Person[] | null => {
+    if (orderParam) {
+      return listOfPeople().sort((a, b) => sortPerson(a, b)).reverse();
+    }
+
+    if (sortParam) {
+      return listOfPeople().sort((a, b) => sortPerson(a, b));
+    }
+
+    return peopleList;
+  };
+
+  const addParentsData = (arrayOfPeople: Person[] | undefined) => {
+    if (!arrayOfPeople) {
+      return null;
+    }
+
+    return arrayOfPeople.map((item) => {
+      const person = { ...item };
+
+      if (item.fatherName || item.motherName) {
+        for (let i = 0; i < listOfPeople().length; i += 1) {
```

Comment:
You don't need to call `listOfPeople()` multiple times. Just create variables
`cont list = listOfPeople();` and work with `list`